python/nano/src/bigdl/nano/diffusion/diffusers/pipelines/stable_diffusion.py

```diff
@@ -0,0 +1,237 @@
+    k_dpm_2_a=diffusers.KDPM2AncestralDiscreteScheduler,
+    k_dpmpp_2=diffusers.DPMSolverMultistepScheduler,
+    k_euler=diffusers.EulerDiscreteScheduler,
+    k_euler_a=diffusers.EulerAncestralDiscreteScheduler,
+    k_heun=diffusers.HeunDiscreteScheduler,
+    k_lms=diffusers.LMSDiscreteScheduler,
+    plms=diffusers.PNDMScheduler,
+)
+default_scheduler = 'k_lms'
+
+
+def _preload_ov(pretrained_model_name_or_path, device, precision):
+    from bigdl.nano.diffusion.diffusers.modules import NanoUNet
+    unet = load_optimized_unet(
+        cache_dir=os.path.join(pretrained_model_name_or_path, 'unet'),
+        device=device, precision=precision)
+    wrapped_unet = NanoUNet(unet)
+    vae_decoder = load_optimized_vae_decoder(
+        cache_dir=os.path.join(pretrained_model_name_or_path, 'vae'),
+        device=device, precision=precision)
+    return wrapped_unet, vae_decoder
+
+
+def _postload_ov(pipe, unet, value_decoder):
+    # load unet and vae
+    pipe.unet = unet
+    pipe.vae.decoder = value_decoder
+
+
+def _load_ipex(pipe, device, precision):
+    if precision == 'float32':
+        vae = InferenceOptimizer.trace(
+            pipe.vae, device=device,
+            use_ipex=True, channels_last=True)
+        text_encoder = InferenceOptimizer.trace(
+            pipe.text_encoder, device=device,
+            use_ipex=True, channels_last=True)
+        unet = InferenceOptimizer.trace(
+            pipe.unet, device=device,
+            use_ipex=True, channels_last=True)
+    elif precision == 'bfloat16':
+        vae = InferenceOptimizer.quantize(
+            pipe.vae, device=device, use_ipex=True,
+            precision="bf16")
+        text_encoder = InferenceOptimizer.quantize(
+            pipe.text_encoder, device=device,
+            use_ipex=True, precision="bf16")
+        unet = InferenceOptimizer.quantize(
+            pipe.unet, device=device,
+            use_ipex=True, precision="bf16")
+    elif precision == 'float16':
+        vae = InferenceOptimizer.quantize(
+            pipe.vae, device=device,
+            use_ipex=True, channels_last=True,
+            precision="fp16")
+        text_encoder = InferenceOptimizer.quantize(
+            pipe.text_encoder, device=device,
+            use_ipex=True, channels_last=True,
+            precision="fp16")
+        unet = InferenceOptimizer.quantize(
+            pipe.unet, device=device,
+            use_ipex=True, channels_last=True,
+            precision="fp16")
+    else:
+        raise ValueError(f'Unsupported precision {precision}, \
+                         available options are: {["float32", "bfloat32", "float16"]}')
+    pipe.vae = vae
+    pipe.text_encoder = text_encoder
+    pipe.unet = unet
+
+
+class NanoDiffusionPipeline:
+    """NanoDiffusionPipeline."""
+
+    @staticmethod
+    def from_pretrained(
+            cls: DiffusionPipeline,
+            pretrained_model_name_or_path,
+            scheduler=default_scheduler,
+            device='iGPU',
```

Comment:
change default device to CPU

Reply:
by default we support IPEX+CPU, change device, precision and backend accordingly.

Reply:
please verify your default config can run successfully on SPR. And for those CPU without bfloat16 support, e.g. ICX, please also verify which precision configuration works.